plugins/pipewire/PipeWireDevice.h

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright 2009-2024 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#pragma once
+
+#ifdef PIPEWIRE_PLUGIN
+#define AUD_BUILD_PLUGIN
+#endif
+
+/**
+ * @file PipeWireDevice.h
+ * @ingroup plugin
+ * The PipeWireDevice class.
+ */
+
+#include <condition_variable>
+#include <thread>
+#include <pipewire/pipewire.h>
+#include <spa/utils/ringbuffer.h>
+
+#include "devices/SoftwareDevice.h"
+
+AUD_NAMESPACE_BEGIN
+
+/**
+ * This device plays back through PipeWire, the simple direct media layer.
+ */
+class AUD_PLUGIN_API PipeWireDevice : public SoftwareDevice
+{
+private:
+	class PipeWireSynchronizer : public DefaultSynchronizer
+	{
+		PipeWireDevice* m_device;
+		int64_t m_tick_start = 0.0f;
+		double m_seek_pos = 0.0f;
+		double m_timeline_pos = 0.0f;
```

Comment:
Just with the pulseaudio changes, I'm pretty sure it should be possible to do everything with just two variables here instead of three. But it's not critical to get this patch in.